tests/Doctrine/Tests/ORM/Functional/Ticket/Issue7052Test.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Tests\Models\CMS\CmsEmployee;
+
+/**
+ * @group issue-7052
+ */
+class Issue7052Test extends \Doctrine\Tests\OrmFunctionalTestCase
+{
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->_schemaTool->createSchema(array(
+            $this->_em->getClassMetadata(__NAMESPACE__ . '\\Issue7052Child'),
+            $this->_em->getClassMetadata(__NAMESPACE__ . '\\Issue7052Parent'),
+        ));
+    }
+
+    public function testIssue()
+    {
+        $parent = new Issue7052Parent();
+        $parent->name = "parent test";
+
+        $childA = new Issue7052Child();
+        $childA->name = "child A";
+        $childA->parent = $parent;
+
+        $childB = new Issue7052Child();
+        $childB->name = "child B";
+        $childB->parent = $parent;
+
+        $this->_em->persist($parent);
+        $this->_em->persist($childA);
+        $this->_em->persist($childB);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $childA = $this->_em->find('Doctrine\Tests\ORM\Functional\Ticket\Issue7052Child', $childA->id);
+
+        $this->_em->clear();
+
+        $childB = $this->_em->find('Doctrine\Tests\ORM\Functional\Ticket\Issue7052Child', $childB->id);
```

Comment:
Can you add an assertion that verifies that this entity is different from the one created before `clear()`? A new variable name is needed.

Reply:
Done: new variable name and assertion to check that they are indeed different objects.
By the way, on our project where we have this issue, the creation and data persistence is done in a different request than the one reading the data. So totally different memory for the PHP process.